mailbox/store/src/main/java/org/apache/james/mailbox/store/AnnotationKeyValidator.java

```diff
@@ -0,0 +1,56 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.store;
+
+import java.util.regex.Pattern;
+
+import org.apache.james.mailbox.exception.MailboxException;
+
+public class AnnotationKeyValidator {
+    public static final Pattern NAME_ANNOTATION_PATTERN = Pattern.compile("[\u0013-\u0024\u0026-\u0029\u002b-\u007F]+");
+
+    public static final String SLASH_CHARACTER = "/";
+
+    public static final String TWO_SLASH_CHARACTER = "//";
+
+    public static boolean isValid(String input) throws MailboxException {
+        if (null == input) {
+            throw new MailboxException("Cannot parse null to a key");
```

Comment:
Maybe a more specific subclass of MailboxException can capture more knowledge about this.

It can have for instance : 
 - a field holding the invalid key value
 - a field holding the cause (why it is rejected)